src/core/common/utils.ts

```diff
@@ -621,3 +621,7 @@ export function compareMarkAttributeNames(a: string, b: string) {
 export function refineColumnName(name: string) {
   return name.replace(/[^0-9a-zA-Z\_]/g, "_");
 }
+
+export function getTimeZoneOffset(date: number) {
+  return new Date(date).getTimezoneOffset() * 60 * 1000;
```

Comment:
Do you need this `date` parameter?  `getTimezoneOffset` only returns the local users offset from GMT, the date doesn't matter here, it is basically ignored.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getTimezoneOffset

So, I'd suggest, if you are just trying to get the local users timezone offset, I would just make this a const something like.

`
export const USER_TIMEZONE_OFFSET = new Date().getTimezoneOffset() * 60 * 1000;
`

or if you are actually trying to get the offset of the actual date number passed in, you're going to need to use a string, otherwise the timezone information is lost.

Reply:
Actually, had to re-read the docs, so the **date** parameter is used, but it converts the offset to the user's local timezone, so,

```js
new Date("2007-12-05T12:30").getTimezoneOffset() == 480
new Date("2007-04-05T12:30").getTimezoneOffset() == 420
```

The first one is PST the second is PDT, since I am in the pacific time zone.

So the comment still stands depending on what you are trying to do.

Reply:
It seems [utcFormat of d3](https://github.com/d3/d3-time-format#locale_utcFormat) what I need